src/Symfony/Component/Security/Guard/Authenticator/GuardAuthenticatorInterface.php

```
@@ -0,0 +1,168 @@
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+use Symfony\Component\Security\Guard\GuardAuthenticatorInterface as GuardAuthenticatorLegacyInterface;
+use Symfony\Component\Security\Guard\Token\GuardTokenInterface;
+
+/**
+ * The interface for all "guard" authenticators.
+ *
+ * The methods on this interface are called throughout the guard authentication
+ * process to give you the power to control most parts of the process from
+ * one location.
+ *
+ * @author Ryan Weaver <ryan@knpuniversity.com>
+ * @author Amaury Leroux de Lens <amaury@lerouxdelens.com>
+ */
+interface GuardAuthenticatorInterface extends GuardAuthenticatorLegacyInterface
+{
+    /**
+     * Does the authenticator supports the given Request ?
+     *
+     * If this returns true, authentication will continue (e.g. getCredentials() will be called).
+     * If false, this authenticator is done. The next (if any) authenticators will be called and
+     * may authenticate the user, or leave the user as anonymous.
+     *
+     * @param Request $request
+     *
+     * @return bool
+     */
+    public function supports(Request $request);
+
+    /**
+     * Get the authentication credentials from the request and return them
+     * as any type (e.g. an associate array). If you return null, authentication
+     * will be skipped.
+     *
+     * Whatever value you return here will be passed to getUser() and checkCredentials()
+     *
+     * For example, for a form login, you might:
+     *
+     *      return array(
+     *          'username' => $request->request->get('_username'),
+     *          'password' => $request->request->get('_password'),
+     *      );
+     *
+     * Or for an API token that's on a header, you might use:
+     *
+     *      return array('api_key' => $request->headers->get('X-API-TOKEN'));
+     *
+     * @param Request $request
+     *
+     * @return mixed|null
+     */
+    public function getCredentials(Request $request);
```

Personally I'd lean more towards `getUser()`, this because it's rather confusing with the Repositories otherwise. By default those start with `find` and it might confuse developers when those names start getting mixed up. Ideally I'd throw an exception in a `get*` method if unable to get whatever was expected. 

How would that work out if the could would always throw an exception if the user could not be found?